modules/common/adapters/point_cloud_adapter.h

```diff
@@ -0,0 +1,304 @@
+/******************************************************************************
```

Comment:
can we hold off this and discuss it tomorrow? I'd like to avoid duplicate code if possible.